spec/src/main/asciidoc/metrics.asciidoc

```diff
@@ -0,0 +1,178 @@
+the methods annotated with a `@Retry`, `@Timeout`, `@CircuitBreaker` or `@Bulkhead` annotation.
+
+=== Names
+
+The automatically added metrics follow a consistent pattern which includes the fully qualified name of the annotated method.
+In the tables below, the placeholder `<name>` should be replaced by the fully qualified method name with all `.` characters replaced with `_`.
+
+=== Metrics added for `@Retry`
+
+[cols="8,3,4,9"]
+|===
+| Name | Type | Unit | Description
+
+|`ft_<name>_retry_calls_success_immediate_total`
+| Counter | None
+| The number of times the method was called and succeeded without retrying
+
+|`ft_<name>_retry_calls_success_retry_total`
+| Counter | None
+| The number of times the method was called and succeeded after retrying at least once
+
+|`ft_<name>_retry_calls_failure_total`
+| Counter | None
+| The number of times the method was called and ultimately failed after retrying
+
+|`ft_<name>_retry_retries_total`
+| Counter | None
+| The total number of times the method was retried
+|===
+
+=== Metrics added for `@Timeout`
+
+[cols="8,3,4,9"]
+|===
+| Name | Type | Unit | Description
+
+|`ft_<name>_timeout_duration`
+| Histogram | Nanoseconds
+| Histogram of execution times for the method
+
+|`ft_<name>_timeout_calls_timeout_true_total`
+| Counter | None
+| The number of times the method timed out
+
+|`ft_<name>_timeout_calls_timeout_false_total`
+| Counter | None
+| The number of times the method completed without timing out
+|===
+
+=== Metrics added for `@CircuitBreaker`
+
+[cols="8,3,4,9"]
+|===
+| Name | Type | Unit | Description
+
+|`ft_<name>_circuitbreaker_state`
+| Gauge<Integer> | None
+| Current state of the circuit breaker. `0` = Closed, `1` = Half-open, `2` = Open.
+
+|`ft_<name>_circuitbreaker_calls_success_total`
+| Counter | None
+| Number of calls allowed to run by the circuit breaker that returned successfully
+
+|`ft_<name>_circuitbreaker_calls_failure_total`
+| Counter | None
+| Number of calls allowed to run by the circuit breaker that then failed
+
+|`ft_<name>_circuitbreaker_calls_circuit_open_total`
+| Counter | None
+| Number of calls prevented from running by an open circuit breaker
+
+|`ft_<name>_circuitbreaker_accumulated_open`
+| Gauge<Long> | Nanoseconds
+| Amount of time the circuit breaker has spent in open state
+
+|`ft_<name>_circuitbreaker_accumulated_half_open`
+| Gauge<Long> | Nanoseconds
+| Amount of time the circuit breaker has spent in half-open state
+
+|`ft_<name>_circuitbreaker_accumulated_closed`
+| Gauge<Long> | Nanoseconds
+| Amount of time the circuit breaker has spent in closed state
+
+|`ft_<name>_circuitbreaker_opened_total`
+| Counter | None
+| Number of times the circuit breaker has moved from closed state to open state
+|===
+
+=== Metrics added for `@Bulkhead`
+
+[cols="8,3,4,9"]
+|===
+| Name | Type | Unit | Description
+
+|`ft_<name>_bulkhead_concurrent_executions`
```

Comment:
Maybe rename to `ft_<name>_bulkhead_concurrent_executions_current` - suffix with `current` to indicate that it's an actual state and not a counter or total.

Reply:
Decided not to do this.